backend/controllers/course_controller.py

```diff
@@ -202,30 +215,53 @@ def update_course(create_course_req: UpdateCourseReq, c_id: str) -> Course:
                 status_code=500, detail=f"Failed to update course: {str(e)}"
             )
 
-
-def get_all_tags(course_id: str) -> CourseTagsResponse:
+def get_all_tags(course_id: str, nested: bool) -> CourseTagsResponse:
+    c_uuid = UUID(course_id)
+    try:
+        tag_counts = count_all_tags(course_id)
+        tags = build_tag_tree(c_uuid) if nested else build_flat_tag_array(c_uuid)
+        tags_list = CourseTagsResponse.model_validate({
+            "tags": tags,
+            "count": tag_counts,
+            })
+        return tags_list
+    except ValidationError as e:
+        logger.error(f"TAGS gotten from DB does not match schema - fix ASAP {str(e)}")
```

Comment:
I'm not sure we need this logger.error line because in the future it would be better to remove and just include the exception, but only a thought. @armintalaie